src/tests/zzzzaccommodation.test.js

```diff
@@ -0,0 +1,137 @@
+import db from '../database/models';
+
+
+import EncryptPassword from '../helpers/Encryptor';
+
+
+dotenv.config();
+
+chai.use(chaiHttp);
+chai.should();
+
+let token;
+let token2;
+const rate = { rate: 3 };
+describe('Accommodation tests', () => {
+  before(async () => {
+    await db.user.destroy({ where: {}, force: true });
+    await db.booking.destroy({ where: {}, force: true });
+    await db.accomodation.destroy({ where: {}, force: true });
+    await db.locations.destroy({ where: {}, force: true });
+    await db.accomodationtype.destroy({ where: {}, force: true });
+    await db.rooms.destroy({ where: {}, force: true });
+    const user = await db.user.create({
+      firstName: 'shema',
+      lastName: 'eric',
+      email: 'shemad@gmail.com',
+      gender: 'male',
+      country: 'Rwanda',
+      birthdate: '12-04-1996',
+      phoneNumber: '0785571790',
+      password: EncryptPassword('shemaeric'),
+      isVerified: true,
+      role: 'manager'
+    });
+    token2 = GenerateToken({ email: 'shemad@gmail.com', isVerified: true, id: user.id });
+    await user.update({ token: token2 });
+
+    const usera = await db.user.create({
+      firstName: 'shema',
+      lastName: 'eric',
+      email: 'shemadd@gmail.com',
+      gender: 'male',
+      country: 'Rwanda',
+      birthdate: '12-04-1996',
+      phoneNumber: '0785571790',
+      password: EncryptPassword('shemaeric'),
+      isVerified: false
+    });
+    token = GenerateToken({ email: 'shemadd@gmail.com', isVerified: false, id: usera.id });
+    await usera.update({ token });
+
+    await db.locations.create({
+      id: 100,
+      city: 'Nairobi'
+    });
+    await db.locations.create({
+      id: 200,
+      city: 'Kampala'
+    });
+    await db.accomodation.create({
+      id: 100,
+      name: 'marriot',
+      description: 'very good',
+      locationId: 100,
+      category: 'hotel',
+      owner: 'jordan',
+      image: 'image'
+    });
+    await db.accomodation.create({
+      id: 200,
+      name: 'serena',
+      description: 'very good',
+      locationId: 200,
+      category: 'hotel',
+      owner: 'jordan',
+      image: 'image'
+    });
+    await db.accomodationtype.create({
+      id: 100,
+      name: ''
+    });
+    await db.usermanagement.create({
+      userId: user.id,
+      managerId: user.id
+    });
+    await db.rooms.create({
+      id: 100,
+      accomodationId: 200,
+      typeId: 100,
+      locationId: 200
+    });
+    await db.booking.create({
+      id: 100,
+      userid: user.id,
+      accommodationid: 200,
+      roomid: 100
+
+    });
+  });
+
+  it('should rate accommodation that is booked', (done) => {
+    chai.request(app).post('/api/v1/accommodations/rate/200')
+      .set('token', `Bearer ${token2}`)
+      .send(rate)
+      .end((err, res) => {
+        res.should.have.status(201);
+        res.body.should.be.an('object');
+        chai.expect(res.body.message).to.eq('You rated this accomodation successfully');
```

Comment:
@mujohn26 When testing, focus on more critical aspects. There's no need of asserting the type of respose.body if you've tested the response status (which is more important than testing it's data type). Similarly, There's no need to test the response message if you test the values in the response object. For test cases where you create/get certain values, test also if the returned values are what you were expecting. You can test statuses and response messages when testing methods (update and delete) that normally do not return results. Kindly update in your tests where the above is applicable.

Reply:
I worked on them all the tests.